x-pack/plugins/alerting/server/saved_objects/index.ts

```diff
@@ -22,6 +22,9 @@ import { isRuleExportable } from './is_rule_exportable';
 import { RuleTypeRegistry } from '../rule_type_registry';
 export { partiallyUpdateAlert } from './partially_update_alert';
 
+// Use caution when removing items from this array! If the field
+// exists in the rule SO, it needs to be included in this array to
+// prevent decryption failures during migration.
```

Comment:
```suggestion
// Use caution when removing items from this array! Any field which has
// ever existed in the rule SO must be included in this array to prevent
// decryption failures during migration.
```

Reply:
Sigh, this system is too brittle.
We should really work with the Kibana team to see if we can make this harder to break and corrupt.